File: src/Gang/Commands/GangCommandWrapper.cs
namespace Gang.Commands
{
    public sealed class GangCommandWrapper
    {
        public GangCommandWrapper(
            string type,
            object data,
            uint? rsn = null
            )
        {
            Type = type;
            Data = data;
            Rsn = rsn;
        }

        public string Type { get; }
        public object Data { get; }
        public uint? Rsn { get; }
    }
}

File: src/GangJs/Core/src/models/communications/GangCommandWrapper.ts
import { IGangCommand } from './IGangCommand';

/** wraps command data */
export class GangCommandWrapper<T> implements IGangCommand {
  constructor(
    /** command data type */
    public readonly type: string,
    /** command data */
    public readonly data: T,
    /** sequence number */
    public readonly sn: number = undefined,
    /** in reply to sequence number */
    public readonly rsn: number = undefined
  ) {}
}
